bats/tests/containers/switch-engines.bats

```diff
@@ -0,0 +1,54 @@
+# Test case 20
+
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start container engine' {
+    start_container_engine
+    wait_for_container_engine
+}
+
+@test "verify we're on docker" {
+    engineName="$(rdctl list-settings | jq -r .containerEngine.name)"
+    case $engineName in
+    containerd)
+        rdctl set --container-engine.name moby
+        wait_for_container_engine
+        ;;
+    esac
+}
+
+@test "pull rancher" {
+    docker run --privileged -d --restart=no -p 8080:80 -p 8443:443 rancher/rancher
+    run docker images --format '{{json .Repository}}
+    assert_output --partial "rancher/rancher"
+}
+
+@test "switch to containerd" {
+    rdctl set --container-engine.name=containerd
+    wait_for_container_engine
+    nerdctl run --privileged -d --restart=no -p 8080:80 -p 8443:443 rancher/rancher
+    run nerdctl images --format '{{json .Repository}}
+    assert_output --partial "rancher/rancher"
+}
+
+@test 'switch back to moby' {
+    rdctl set --container-engine.name moby
+    wait_for_container_engine
+}
+
+@test 'verify the rancher container is gone' {
+    run docker images --format '{{json .Repository}}
+    refute_output --partial "rancher/rancher"
```

Comment:
Does this actual work? I don't think we delete the image store when we are switching engines, so I would expect the old images to still be there when we are switching back.

I agree that we should at least offer the option to delete images when switching, so not to trap diskspace, but we don't do this yet.

Reply:
The key to switching engines in bats is to remember to set `RD_CONTAINER_ENGINE` each time. This isn't ideal.

Reply:
> The key to switching engines in bats is to remember to set RD_CONTAINER_ENGINE each time. This isn't ideal.

I agree, we should refactor this, but not in this PR.